Library/ViewModels/RewardsCollectionViewModel.swift

```diff
@@ -51,23 +52,28 @@ public final class RewardsCollectionViewModel: RewardsCollectionViewModelType,
 
     self.selectedRewardProperty <~ selectedRewardFromId
 
-    self.goToPledge = Signal.combineLatest(
-      self.configureWithProjectProperty.signal.skipNil(),
+    let refTag = configData
+      .map(second)
+
+    let goToPledge = Signal.combineLatest(
+      project,
       selectedRewardFromId,
-      self.configureWithRefTagProperty.signal
+      refTag
     )
     .map { project, reward, refTag in
       PledgeData(project: project, reward: reward, refTag: refTag)
     }
+
+    self.goToPledge = goToPledge
+      .filter { _ in featureNativeCheckoutPledgeViewEnabled() }
+
+    self.goToDeprecatedPledge = goToPledge
+      .filter { _ in !featureNativeCheckoutPledgeViewEnabled() }
   }
 
-  private let configureWithProjectProperty = MutableProperty<Project?>(nil)
-  private let configureWithRewardsProperty = MutableProperty<[Reward]?>(nil)
-  private let configureWithRefTagProperty = MutableProperty<RefTag?>(nil)
+  private let configDataProperty = MutableProperty<(Project, RefTag?)?>(nil)
```

Comment:
Just brought this back to being one `MutableProperty` and moved the splitting out into the signal definitions.